### DISCUSSION TITLE
How to compute the elasticity tensor of orthotropic symmetry material for a curved shell?

### DISCUSSION THREAD

**Discussion by in55555** [idaholab/moose#29118] [Q&A Modules: Solid mechanics]
Hello,

I want to do the mechanical simulation for a curved carbon fiber reinforced polymer shell. And I have used the ComputeElasitictyTensor to compute its elasiticity tensor. However, I found that this syntax can only rotate the tensor once, but can not always rotate it following the curve. Is there any syntax or options can realize of this function?

I'm a newcomer for moose and I'm not familiar its functions yet. I know this question is stupid, but I really need you guys help.

Thank you! And this is the mesh for the shell.
![无标题](https://github.com/user-attachments/assets/de8effe7-cbf3-4f03-97e0-65a74258d714)

**Comment by GiudGiud** [comment]
Hello

So are you using 2D shell elements then? Or thin 3D elements?
And I assume the elasticity tensor is not isotropic since you need to rotate it?

@gambka do we have this in the module? An anisotropic tensor with a couple variables or functions to describe the local frame axis?

**Reply by gambka** [reply]
@GiudGiud your question on whether it is shell element or solid elements is valid. The models for mechanics are different depending on the element type. If shell elements are being used, I'm not that familiar with the shell mechanical models implemented into the solid mechanics module, @bwspenc would know more on that topic. For the solid element (thin Hex elements) we do have anisotropic capabilities for elasticity, plasticity, and creep. However, I think as @in55555 mentioned the rotation is done once. I am not aware of capabilities in the solid_mechanics module for the local rotation along the curve. 

If I recall correctly, @jasondhales has done work for local rotation for mechanical problems, but it may for a internal application. Perhaps that functionality could be moved into MOOSE, but he would have to comment on that.

**Reply by in55555** [reply]
Yes, the tensor is anisotropic. I was trying to use a shell element, but I didn't find a anisotropic option for shell elements, but only a ([ADComputeIsotropicElasticityTensorShell](https://mooseframework.inl.gov/source/materials/ADComputeIsotropicElasticityTensorShell.html) for isotropic shell. So I choose the ComputeElasitictyTensor for 3D solid elements, but this syntax can only rotate once and can not always rotate following the curve.
